hazardous/survtrace/_encoder.py

```diff
@@ -0,0 +1,215 @@
+import numpy as np
```

Comment:
Please add a header to note that this code is a refactoring derived from
https://github.com/RyanWangZf/SurvTRACE also published under the MIT license
with the following copyright:

```
Copyright (c) 2021 Maximum
```